.github/workflows/e2e-upgrade.yaml

```diff
@@ -0,0 +1,72 @@
+name: E2E Upgrade tests
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - starship/**
+  pull_request:
+    branches:
+      - main
+    paths:
+      - starship/**
+
+jobs:
+  e2e-v7v8-test:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}
+      cancel-in-progress: ${{ github.event_name == 'pull_request' }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up Go 1.x
+        uses: actions/setup-go@v2
+        with:
+          go-version: ^1.19
+        id: go
+
+      - name: Set namespace
+        id: set-namespace
+        run: |
+          namespace="ci-${{ github.repository_owner }}-${{ github.workflow }}-${{ github.ref }}"
+          echo "Setting namespace to $namespace"
+          echo "namespace=$namespace" >> $GITHUB_OUTPUT
+
+      # Starship Infra setup
+      # - Connects to k8s cluster with kubeconfig (digital ocean)
+      # - Creates a new namespace based on the name (deleted in next step)
+      # - Spins up the infra with the given config file
+      # - Waits till all nodes are running (timeout 30m)
+      # - Port forward all ports to localhost for next steps to connect
+      - name: Setup Test infra
+        id: starship-action
+        uses: cosmology-tech/starship-action@0.2.9
+        with:
+          values: starship/v7v8/config.yaml
+          port-forward: true
+          version: 0.1.23
+          kubeconfig: ${{ secrets.KUBECONFIG }}
+          namespace: ${{ steps.set-namespace.outputs.namespace }}
+
+      - name: Run Tests
+        run: |
+          cd starship/v7v8/ 
+          go mod tidy
+          make test
+
+      # Starship resource cleanup on cluster
+      # todo(@anmol1696): change this to be post step of the action
```

Comment:
Is this still in WIP?

Reply:
It is a todo, currently this step will work as we need it to, i just dont think it is super clean.

Once I have a new version of the starship-action, i will come back and fix this as well, hence the todo here.

This should not be blocking though.